### DISCUSSION TITLE
Inconsistent kwargs key in `auth.py` and the key checked in auth_manager's `get_url_login` method

### DISCUSSION THREAD

**Discussion by awdavidson** [apache/airflow#37224] [General]
We have implemented a webserver_config that adds functionality to ensure an unauthenticated request to an endpoint is redirected back to the original location once the user has authenticated. For example navigating to `http://127.0.0.1:8080/dags/example_bash_operator/grid` in a browser without logging in will redirect the user to the login page `http://127.0.0.1:8080/login` then redirect back to `http://127.0.0.1:8080/dags/example_bash_operator/grid` once the user has successfully logged in.

After upgrading from `2.6.3` to `2.7.3` this functionality broke; during my investigation I noticed the following change introduced a change to authentication managers: https://github.com/apache/airflow/pull/32697

This change, has also introduced inconsistency with kwargs set in `auth.py` and the kwargs checked in the auth managers `get_url_login` method meaning the `next` URL is no longer passed through to the redirected request so a user is no longer redirected back to the original URL.

e.g.
`get_url_login` checks kwargs for `next_url` https://github.com/aws-mwaa/upstream-to-airflow/blob/333a0527430b3dde7cbedaf407560cebffc20827/airflow/auth/managers/fab/fab_auth_manager.py#L58 but `auth.py` sets `next` https://github.com/aws-mwaa/upstream-to-airflow/blob/333a0527430b3dde7cbedaf407560cebffc20827/airflow/www/auth.py#L64

I am under the impression this is a bug? Either the kwargs need to be consistent or the auth manager needs to check both `next_url` and `next`keys?

Do others agree?

NB: I am happy to submit a PR with the change if required

**Comment by potiuk** [comment]
cc: @vincbeck - - we are likely going to release a new Airflow Release shortly, so we might have a chance to get this one in if we confirm/fix it

**Comment by vincbeck** [comment]
Thanks @awdavidson for reporting this bug and sharing all the details, it makes the debugging very easy :) Indeed, there is a bug here and we should make the parameter consistent, we should use `next_url` everywhere. I can submit a PR to fix it

**Reply by vincbeck** [reply]
Fix: #37225

**Reply by awdavidson** [reply]
Thank you for the quick turn around.

Do such fixes get backported? Or only fixed forwarded requiring an upgrade?

**Reply by vincbeck** [reply]
Latter :)

**Comment by vincbeck** [comment]
Closing the discussion since the fix has been merged. Thanks again for reporting this bug. The fix will be part of Airflow 2.8.2
